apis/logging/v1/input_source_types.go

```diff
@@ -0,0 +1,52 @@
+package v1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+)
+
+// NOTE: The Enum validation on SourceSpec.Type must be updated if the list of types changes.
+
+// SourceSpec is a union of input source types.
+//
+// The fields of this struct define the set of known source types.
+type SourceSpec struct {
+	HTTP *HTTPSource `json:"http,omitempty"`
+}
+
+const (
+	FormatK8S  = "k8s"  // Log events in k8s list format, e.g. API audit log events.
+	FormatText = "text" // Newline-separated plain-text log records.
+)
+
+// HTTPSource receives encoded logs as a HTTP endpoint.
+type HTTPSource struct {
+
+	// FIXME TLS server configuration - can this be handled in the Service,
+	// or do we need vector to have certificates? Can the service be bypassed?
+
+	// FIXME What to do about service name clashes? Can't generate random name, service name is used to locate the source.
+
+	// Name of the source.
+	// A Service resource is generated for each source,
+	// the name must be a unique Service name in the ClusterLogForwarder's namespace.
+	//
+	// +required
+	Name string `json:"name"`
+
+	// Port that this source will listen on.
+	// Used to create a Service for this port.
+	//
+	// +required
+	Port corev1.ServicePort `json:"port"`
+	// Format is the format of incoming log data.
+	//
+	// +kubebuilder:validation:Enum:=k8s;text
+	// +required
+	Format string `json:"format"`
```

Comment:
My previous comment from the other PR:
* we don't actually support "text" based on what I have seen in the rest of the impl so we should drop it or further define what "text" means.  I assume it would be a broad category
* Do we need to expand what k8s means?  I know its a list of "kubernetes events" but k8s is not very specific.  What if we get a list of pod yaml?  shoudnt we be more specific given we are only going to accept a list of v1.Events?

Reply:
@jcantrill audit events

Reply:
Aren't 'audit events' and corev1.Events the same?

Reply:
Nope, check https://kubernetes.io/docs/tasks/debug/debug-cluster/audit/  out. https://kubernetes.io/docs/reference/kubernetes-api/cluster-resources/event-v1/ vs https://kubernetes.io/docs/reference/config-api/apiserver-audit.v1/#audit-k8s-io-v1-Event.